### PR TITLE
Tezos 9.2

### PR DESCRIPTION
**Pull request by pirbo**
This is a "minor" release. It adds the primitives and binaries of for a new "economic protocols" for Tezos: **Granada**

### CODE REVIEW

**Comment by pmetzger** [comment]
FYI, I think you misspelled "Granada".

File: packages/tezos/tezos.9.2/opam [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+]
+synopsis: "Tezos meta package installing all active binaries"
+
+url {
+  src: "https://gitlab.com/tezos/tezos/-/archive/v9.1/tezos-v9.1.tar.bz2"
+  checksum: [
+    "sha256=114073c8038a3329158557cfec00d2c8a1fadac3b622b30f8ecf1479949028d0"
+    "sha512=79631f68ac61c139ff849f23b6a9301c090d6afec091bde855e05ec753151a0259d61e509b67162ff49a2c80872d5a1d1ec65ff32b5cf09b6ea4684e269e4fd7"
+  ]
+}
```
Comment (kit-ty-kate) [inline packages/tezos/tezos.9.2/opam:49]:
```suggestion
}
```
This is the archive for tezos 9.1. Furthermore I don't think having an archive is required for this package

Comment (pirbo) [reply packages/tezos/tezos.9.2/opam]: I fully agree but I put them to turn of the warning "non conf packages with no sources" (or something similar) :-)

Comment (avsm) [reply packages/tezos/tezos.9.2/opam]: Let's try to go with accurate opam metadata please -- we'll fix the CI. It exists to help us humans, not the other way around :-)

**Comment by kit-ty-kate** [comment]
Thanks! Could you have a look at the tests failure in tezos-crypto (I believe it was also failing in the last release)
```
#=== ERROR while compiling tezos-crypto.9.2 ===================================#
# context              2.1.0~beta4 | linux/x86_64 | ocaml-base-compiler.4.09.1 | pinned(https://gitlab.com/tezos/tezos/-/archive/v9.2/tezos-v9.2.tar.bz2)
# path                 ~/.opam/4.09/.opam-switch/build/tezos-crypto.9.2
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune runtest -p tezos-crypto -j 31
# exit-code            1
# env-file             ~/.opam/log/tezos-crypto-1968-b0794b.env
# output-file          ~/.opam/log/tezos-crypto-1968-b0794b.out
### output ###
# test_prop_hacl_hash alias src/lib_crypto/test/runtest
# SHA256_incremental_one: PASS
# 
# SHA256_incremental_list: PASS
# 
# SHA512_incremental_one: PASS
# 
# SHA512_incremental_list: PASS
# 
# test_signature alias src/lib_crypto/test/runtest
# Testing `signature'.
# This run has ID `6C1CF264-6C9F-4B8E-A436-A5FB91AB220B'.
# 
#   [OK]          size          0   size.
#   [OK]          size          1   test_of_bytes_without_validation.
# 
# Full test results in `~/.opam/4.09/.opam-switch/build/tezos-crypto.9.2/_build/default/src/lib_crypto/test/_build/_tests/signature'.
# Test Successful in 0.011s. 2 tests run.
# test_blake2b alias src/lib_crypto/test/runtest
# Testing `tezos-crypto'.
# This run has ID `775AC647-E8D1-4D48-98E9-C740C42211D0'.
# 
#   [OK]          blake2b          0   hash hex/dehex.
#   [OK]          blake2b          1   hash print/parse.
# 
# Full test results in `~/.opam/4.09/.opam-switch/build/tezos-crypto.9.2/_build/default/src/lib_crypto/test/_build/_tests/tezos-crypto'.
# Test Successful in 0.006s. 2 tests run.
#  test_base58 alias src/lib_crypto/test/runtest
# Testing `tezos-crypto'.
# This run has ID `9D3BDC1A-2858-46F6-B842-30A4726E7B39'.
# 
#   [OK]          base58          0   safe decoding.
#   [OK]          base58          1   safe encoding/decoding.
#   [OK]          base58          2   raw encoding/decoding.
# 
# Full test results in `~/.opam/4.09/.opam-switch/build/tezos-crypto.9.2/_build/default/src/lib_crypto/test/_build/_tests/tezos-crypto'.
# Test Successful in 0.049s. 3 tests run.
# test_crypto_box alias src/lib_crypto/test/runtest
# Testing `tezos-crypto'.
# This run has ID `34CDBB17-3B42-4D03-AE39-5F449BFF5929'.
# 
#   [OK]          crypto_box          0   Neuterize Secret roundtrip.
#   [OK]          crypto_box          1   Public Key Hash roundtrip.
#   [OK]          crypto_box          2   HACL* box (noalloc).
#   [OK]          crypto_box          3   HACL* box.
# 
# Full test results in `~/.opam/4.09/.opam-switch/build/tezos-crypto.9.2/_build/default/src/lib_crypto/test/_build/_tests/tezos-crypto'.
# Test Successful in 0.001s. 4 tests run.
#    test_pvss alias src/lib_crypto/test/runtest
# Testing `test-pvss'.
# This run has ID `AE316EC0-AF9D-4439-9CBB-A5776861CE7F'.
# 
#   [OK]          pvss          0   dealer proof.
#   [OK]          pvss          1   invalid dealer proof.
#   [OK]          pvss          2   reveal.
#   [OK]          pvss          3   invalid reveal.
#   [OK]          pvss          4   reconstruct.
#   [OK]          pvss          5   invalid reconstruct.
#   [OK]          pvss          6   randomness commitment protocol.
# 
# Full test results in `~/.opam/4.09/.opam-switch/build/tezos-crypto.9.2/_build/default/src/lib_crypto/test/_build/_tests/test-pvss'.
# Test Successful in 0.129s. 7 tests run.
#  test_merkle alias src/lib_crypto/test/runtest
# Testing `tezos-crypto'.
# This run has ID `4BED52DC-D4C3-45A6-9953-A868942B3B23'.
# 
#   [OK]          merkle          0   compute.
#   [OK]          merkle          1   compute_examples.
#   [OK]          merkle          2   path.
#   [OK]          merkle          3   path_examples.
# 
# Full test results in `~/.opam/4.09/.opam-switch/build/tezos-crypto.9.2/_build/default/src/lib_crypto/test/_build/_tests/tezos-crypto'.
# Test Successful in 0.091s. 4 tests run.
# test_ed25519 alias src/lib_crypto/test/runtest
# Testing `tezos-crypto'.
# This run has ID `9E9C6D03-1B48-44E2-94CF-96CF66550BA4'.
# 
#   [OK]          ed25519          0   b58check.roundtrip.
#   [OK]          ed25519          1   b58check.invalid.
#   [OK]          ed25519          2   b58 pkh encodings.
#   [OK]          ed25519          3   b58 key encodings.
# 
# Full test results in `~/.opam/4.09/.opam-switch/build/tezos-crypto.9.2/_build/default/src/lib_crypto/test/_build/_tests/tezos-crypto'.
# Test Successful in 0.095s. 4 tests run.
#    test_hacl alias src/lib_crypto/test/runtest
# Testing `hacl'.
# This run has ID `1007FC83-2AE9-4421-8F54-74EE0344B775'.
# 
#   [OK]          hash               0   hmac_sha256.
#   [OK]          hash               1   hmac_sha512.
#   [OK]          hash               2   sha256.
#   [OK]          hash               3   sha256_seq.
#   [OK]          hash               4   sha512.
#   [OK]          hash               5   sha3_256.
#   [OK]          hash               6   sha3_512.
#   [OK]          hash               7   keccak_256.
#   [OK]          blake2b            0   0.
#   [OK]          blake2b            1   1.
#   [OK]          blake2b            2   2.
#   [OK]          secretbox          0   secretbox.
#   [OK]          box                0   box.
#   [OK]          ed25519            0   keypair.
#   [OK]          ed25519            1   sign.
#   [OK]          ed25519            2   public.
#   [OK]          p256               0   export.
#   [OK]          p256               1   write_key.
#   [OK]          p256               2   keypair.
#   [OK]          p256               3   sign.
#   [OK]          p256               4   test_vectors.
# 
# Full test results in `~/.opam/4.09/.opam-switch/build/tezos-crypto.9.2/_build/default/src/lib_crypto/test/_build/_tests/hacl'.
# Test Successful in 1.945s. 21 tests run.
#    test_p256 alias src/lib_crypto/test/runtest
# Testing `tezos-crypto'.
# This run has ID `2AC8A10E-E169-40D1-A80D-59CFBDA726F3'.
# 
#   [OK]          p256          0   b58check.roundtrip.
#   [OK]          p256          1   b58check.invalid.
#   [OK]          p256          2   b58 pkh encodings.
#   [OK]          p256          3   b58 key encodings.
# 
# Full test results in `~/.opam/4.09/.opam-switch/build/tezos-crypto.9.2/_build/default/src/lib_crypto/test/_build/_tests/tezos-crypto'.
# Test Successful in 0.233s. 4 tests run.
# test_prop_signature_pk alias src/lib_crypto/test/runtest
# P256_pk_of_bytes: PASS
# 
# test_prop_signature alias src/lib_crypto/test/runtest
# Ed25519_sign_check: PASS
# 
# P256_sign_check: PASS
# 
# Secp256k1_sign_check: PASS
# 
# Signature_Ed25519_sign_check: PASS
# 
# Signature_Secp256k1_sign_check: PASS
# 
# test_deterministic_nonce alias src/lib_crypto/test/runtest (exit 125)
# (cd _build/default/src/lib_crypto/test && ./test_deterministic_nonce.exe)
# Testing `tezos-crypto'.
# This run has ID `2FEF5C7A-F542-4279-9E74-416DC861F4F6'.
# 
#   [OK]          deterministic_nonce          0   hash_matches_ed25519.
# test_deterministic_nonce.exe: internal error, uncaught exception:
#                               Unix.Unix_error(Unix.ENOENT, "open", "/home/opam/.opam/4.09/.opam-switch/build/tezos-crypto.9.2/_build/default/src/lib_crypto/test/_build/_tests/tezos-crypto/deterministic_nonce.001.output")
#                               
```
Full log: https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/8edf08059674cd74992d636f3f35e8c0bd26579a/variant/opam-2.1,compilers,4.09,tezos-crypto.9.2,tests

Could you also return the other few fixes in the opam files uptream if it's not already done?

**Pull request merged**